anima/env/mayaEnv/auxiliary.py

```diff
@@ -914,7 +937,40 @@ def run_post_publishers():
 
         # before running use the staging area to store the current version
         staging['version'] = version
-        run_publishers(type_name, publisher_type=POST_PUBLISHER_TYPE)
+
+        # show splash screen during post publish progress and lock maya
+        here = os.path.normpath(os.path.dirname(os.path.realpath(__file__)))
+        image_path = os.path.normpath(
+            os.path.join(here, 'config', 'images', 'splash_screen.jpg')
+        ).replace('\\', '/')
+
+        p = QtGui.QPixmap(image_path)
+        s = QtWidgets.QSplashScreen(p, QtCore.Qt.WindowStaysOnTopHint)
+        s.setEnabled(False)
+        s.setWindowModality(QtCore.Qt.ApplicationModal)
+        s.show()
+
+        try:
+            run_publishers(type_name, publisher_type=POST_PUBLISHER_TYPE)
+            s.close()
+        except (PublishError, RuntimeError) as e:
+            s.close()
+            # do not forget to clean up the staging area
+            staging.clear()
+            # pop up a message box with the error
+            pm.confirmDialog(
+                title='PublishError',
+                icon='critical',
+                message='<b>%s</b><br/><br/>%s' % (
+                    'POST PUBLISH FAILED!!!',
+                    str(''.join([i for i in unicode(e) if ord(i) < 128]))
```

Comment:
also if you think that you need this code and if you are going to use it again and again stay DRY (Don't Repeat Yourself) and write a helper function may be under ``anima.utils`` for that.